services/custom_fees.proto

```diff
@@ -173,3 +173,15 @@ message AssessedCustomFee {
    */
   repeated AccountID effective_payer_account_id = 4;
 }
+
+message ConsensusCustomFee {
+  /**
+  * Fixed fee to be charged
+  */
+  FixedFee fixed_fee = 1;
+
+  /**
+  * The account to receive the custom fee
+  */
+  AccountID fee_collector_account_id = 2;
+}
```

Comment:
1. Why are we defining a whole new custom fee type here?  Would it not make more sense to use the existing custom fee structure?
1. There must be a newline at EOF.
1. Specification comments should specify behavior of the field and follow correct formatting
1. Messages must have specification comments that describe the use of that message and interactions with other systems (e.g. Block Stream), if any.
```suggestion
/**
 * A custom fee definition for a consensus topic.
 *
 * This fee definition is specific to an Hedera Consensus Service (HCS) topic
 * and SHOULD NOT be used in any other context.<br/>
 * All fields for this message are REQUIRED.<br/>
 * Only "fixed" fee definitions are supported because there is no basis for
 * a fractional fee on a consensus submit transaction.
 */
message ConsensusCustomFee {
    /**
     * A fixed custom fee.
     * <p>
     * The amount of HBAR or other token described by this `FixedFee` SHALL
     * be charged to the transction payer for each message submitted to a
     * topic that assigns this consensus custom fee.
     */
    FixedFee fixed_fee = 1;

    /**
     * A collection account identifier.
     * <p>
     * All amounts collected for this consensus custom fee SHALL be transferred
     * to the account identified by this field.
     */
    AccountID fee_collector_account_id = 2;
}

```
Note: I am guessing a bit as to behavior in this suggestion; please correct any errors above.

Reply:
> 1) Why are we defining a whole new custom fee type here? Wouldn't it make more sense to use the existing custom fee structure?

We tried to recycle as much as possible, and considering the final schema we needed (a fixed fee + an account), we ended up with this definition.

There are two other messages that use FixedFee: CustomFee and RoyaltyFee. RoyaltyFee doesn't fit because of the data type, and CustomFee is too permissive (it accepts values we don't accept for this function).

But the HIP just suggests the proto definition. Do you see other ways we can define this?

Reply:
Most use of `CustomFee` applies point-specific restrictions.  We could allow `CustomFee` and specify that only a `FixedFee` is permitted for Consensus Topics.  This is not dissimilar to how different token types only allow certain types of `CustomFee` to apply.

We can also use the approach detailed here; it's just that we then have to modify a lot more of the system to handle special cases everywhere we might encounter a `ConsensusCustomFee` instead of just reusing all the existing custom fee logic.

I _believe_ it would be clearer, and easier to implement, if we use `CustomFee` for the Consensus Topic custom fees and just document that "This field MUST contain only `FixedFee` variants of `CustomFee`" on the custom fees field for `topicCreate` and `topicUpdate`.

Reply:
Personally, I still prefer a clean dedicated message, but if reusing `CustomFee` is consistent with what we're already doing elsewhere, it works for me.

However, it's important to note that with the same HIP, we're going to create a new allowance transaction type to avoid cross-dependencies between services, so using a dedicated fee message can be seen as consistent with that goal.

We need @netopyr's decision on this.

Reply:
I'll read that HIP for the new transaction type, thanks for surfacing that.

Reply:
Thanks to you! Please wait for the final version of the doc, the definition of the new tx is still missing from the current visible HIP. We will update the HIP text after we defined all the details.